### ISSUE TITLE
pulp upgrade: pre-#1600 versions cannot self-rescue (chicken-and-egg follow-up)

### ISSUE THREAD

**Issue by danielraffel** [danielraffel/pulp#1672]
The fix in #1600 made `pulp upgrade` discovery synchronous, but **users stuck on a pre-#1600 binary cannot upgrade through the buggy upgrade path**. They must manually intervene.

## Reproduction

User installs pulp 0.73.0 (or any pre-v0.78.0 version), tries `pulp upgrade`:

```
$ pulp version
Pulp SDK version: 0.73.0
$ pulp upgrade
Checking for updates...
  Target version: 0.73.0          ← reads stale cache, thinks 0.73 is latest
  Downloading pulp-darwin-arm64.tar.gz...
  ✓ Pulp CLI upgraded to v0.73.0   ← "upgrades" to itself
```

The user is now stuck — every subsequent `pulp upgrade` re-downloads 0.73.0 and never advances. The fix in #1600 only helps users who already have a version that contains it.

## Root cause (recap)

Pre-#1600, `kick_background_refresh` in `pulp_cli.cpp` used `std::thread(...).detach()` for the cache refresh. The detached thread is reaped at process exit before curl finishes, so the cache file never updates. Subsequent invocations read the stale cache and treat `latest_version: 0.73.0` as truth.

## Workarounds for affected users

1. **Clear the cache** (forces synchronous fetch on the buggy binary's empty-cache path):
   ```bash
   rm ~/.pulp/update-cache.json && pulp upgrade
   ```
2. **Explicit version positional arg**:
   ```bash
   pulp upgrade 0.78.2
   ```
3. **Manual download**:
   ```bash
   curl -fsSL https://github.com/danielraffel/pulp/releases/download/v0.78.2/pulp-darwin-arm64.tar.gz \
     | tar -xzC ~/.pulp/bin/
   ```

### A. Install script: detect and self-rescue
The install script at the project's install URL should detect that the user already has a buggy version (any 0.73.x or earlier per the introduction range) and force-overwrite to the latest tar.gz. Single bash one-liner.

### B. Pulp release notes / banner
The next release's banner could mention this — but that requires the user to have already escaped the bug.

### C. Optional: stale-cache age sanity check
Future versions of `pulp upgrade` could detect when `last_check_epoch_sec` is "fresh" (< 24h ago) but the cached `latest_version` is OLDER than the running binary's compiled version — that's an obvious staleness signal that warrants forcing a refresh. But this only helps users with a NEWER buggy version, not users on 0.73.x.

### D. Document workaround prominently
README + install docs should call out the rescue commands so users hit them on Google rather than filing bugs.

## Action items
- [ ] Update install script to overwrite existing pulp binary unconditionally with latest (fast escape)
- [ ] Add the workaround commands to README / docs / a TROUBLESHOOTING.md
- [ ] Consider an optional `pulp upgrade --force` flag that bypasses the cache entirely (cheap, prevents future occurrences)
